packages/canon/src/components/Input/styles.css

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.canon-input-root {
+  display: flex;
+  flex-direction: column;
+  font-family: var(--canon-font-regular);
+  width: 100%;
+}
+
+.canon-input-label {
+  font-size: var(--canon-font-size-caption);
+  font-weight: var(--canon-font-weight-regular);
+  color: var(--canon-text-primary);
+  margin-bottom: var(--canon-spacing-3xs);
+}
+
+.canon-input-control {
+  border-radius: var(--canon-border-radius-sm);
+  border: 1px solid var(--canon-border-base);
+  padding: var(--canon-spacing-sm);
+  background-color: var(--canon-surface-1);
+  font-size: var(--canon-font-size-body);
+  font-weight: var(--canon-font-weight-regular);
+  color: var(--canon-text-primary);
+  transition: border-color 0.2s ease-in-out, outline-color 0.2s ease-in-out;
+}
+
+.canon-input-control::placeholder {
+  color: var(--canon-text-secondary);
+}
+
+.canon-input-control:hover {
+  border-color: var(--canon-border-hover);
+}
+
+.canon-input-control:focus-visible {
+  outline-color: var(--canon-border-selected);
+  outline-width: 0px;
+  border-color: var(--canon-border-selected);
+}
+
+.canon-input-control[data-invalid] {
```

Comment:
QQ: are all of these various things documented somewhere? In the Base UI docs I see what seems to be an alternate `.*--error` pattern instead

Reply:
Was looking at the old docs, this is documented in the new docs: https://base-ui.com/react/components/input